flink-runtime/src/main/java/org/apache/flink/runtime/io/network/metrics/AbstractBuffersUsageGauge.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.io.network.metrics;
+
+import org.apache.flink.annotation.VisibleForTesting;
+import org.apache.flink.metrics.Gauge;
+import org.apache.flink.runtime.io.network.partition.consumer.SingleInputGate;
+
+/**
+ * Abstract gauge implementation for calculate the buffer usage percent.
+ */
+public abstract class AbstractBuffersUsageGauge implements Gauge<Float> {
+
+	protected SingleInputGate[] inputGates;
+
+	@VisibleForTesting
+	public abstract int calculateUsedBuffers(SingleInputGate inputGate);
+
+	@VisibleForTesting
+	public abstract int calculateBufferPoolSize(SingleInputGate inputGate);
```

Comment:
how about renaming `calculateBufferPoolSize` as `calculateTotalBuffers`, because for exclusive case it has not buffer pool, so this naming seems not general.

Reply:
agree